### PR TITLE
docs: simplify workflow-templates descriptions

### PR DESCRIPTION
**Pull request by stvstnfrd**
to remove the boilerplate/superfluous text:

> Use this workflow to ...

This makes it easier to parse the list of organizational workflows
and is in following the convention used by Github's "popular services".

> Publish ...
> Build and test ...
> Setup ...

### CODE REVIEW

**Review by sarina — APPROVED** [review]
I'm not sure I particularly see the need for this, but I also have no problem and see how it's a simplification

**Comment by stvstnfrd** [comment]
Just for some color/context before I merge because admittedly, this is "button" for me...

> I'm not sure I particularly see the need for this, but I also have no problem and see how it's a simplification

I think it's actually a  bigger and more pervasive issue, both in edx's docs, as well as in the industry as a whole.

To be clear, this isn't just me "guessing" what I hypothetically think may be a problem for some;
I personally had difficulty reading+parsing the list of options, so I'm reacting to a problem I had :)

The human brain tends to stop critical processing if/when it encounters duplicative text.
Psychology aside, superfluous text takes longer to read and detracts from the pertinent writing.

This copy/paste + cargo-culting leads to things like unit tests with docstrings like:

> This is a test that aims to check and see if the data is valid
> This is a test that aims to check and see if we have sufficient permissions

instead of:

> Check for valid data
> Check for sufficient permissions

The latter is easier to read and understand, especially when we consider that:
it's a `test_*` method in a `TestCase` in a file named `test_*.py` in a directory called `tests`;
we don't need to reiterate that "this is a test". Similarly, this is a template, in a directory called `workflow-templates`, in the special `.github` repo, that maintainers will get to via the "Add a new workflow" button/workflow. Repeating that they can "use this (github) workflow to..." isn't actually helpful and unhelpful docs should be considered harmful. It's the `user.delete()  # this deletes a user` thing ;)

It's hard enough to get developers to RTFM; reducing boilerplate/superfluousness reduces the burden.

:)

**Pull request merged**
